contributors/design-proposals/api-machinery/alternate-api-representations.md

```diff
@@ -0,0 +1,438 @@
+   support (ask for v2, then v1). The Accept mechanism already handles this sort of
+   multi-version negotiation, while any approach based on query parameters would
+   have to implement this option as well. In addition, some serializations may not
+   be valid in all content types, so the client asking for TabularOutput in protobuf
+   may also ask for TabularOutput in JSON - if TabularOutput is not valid in protobuf
+   the server call fall back to JSON.
+
+*  Use new resource paths - `/apis/autoscaling/v1/namespaces/example/horizontalpodautoscalermetadata`
+
+   This leads to a proliferation of paths which will confuse automated tools and end
+   users. Authorization, logging, audit may all need a way to map the two resources
+   as equivalent, while clients would need a discovery mechanism that identifies a
+   "same underlying object" relationship that is different from subresources.
+
+*  Use a special HTTP header to denote the alternative representation
+
+   Given the need to support multiple versions, this would be reimplementing Accept
+   in a slightly different way, so we prefer to reuse Accept.
+
+*  For partial object retrieval, support complex field selectors
+
+   From an efficiency perspective, calculating subpaths and filtering out sub fields
+   from the underlying object is complex.  In practice, almost all filtering falls into
+   a few limited subsets, and thus retrieving an object into a few known schemas can be made
+   much more efficient. In addition, arbitrary transformation of the object provides
+   opportunities for supporting forward "partial" migration - for instance, returning a
+   ReplicationController as a ReplicaSet to simplify a transition across resource types.
+   While this is not under explicit consideration, allowing a caller to move objects across
+   schemas will eventually be a required behavior when dramatic changes occur in an API
+   schema.
+
+## Backwards Compatibility
+
+### Old clients
+
+Old clients would not be affected by the new Accept path.
+
+If servers begin returning Status in version `meta.k8s.io/v1`, old clients would likely error
+as that group has never been used.  We would continue to return the group version of the calling
+API group on server responses unless the `sv` mime-type parameter is set.
+
+
+### Old servers
+
+Because old Kubernetes servers are not selective about the content type parameters they
+accept, we may wish to patch the 1.3 and 1.4 server versions to explicitly bypass content
+types they do not recognize the parameters to. As a special consideration, this would allow
+new clients to more strictly handle Accept (so that the server returns errors if the content
+type is not recognized).
+
+As part of introducing the new API group `meta.k8s.io`, some opaque calls where we assume the
+empty API group-version for the resource (GET parameters) could be defaulted to this group.
+
+
+## Future items
+
+* ???
+
+
+<!-- BEGIN MUNGE: GENERATED_ANALYTICS -->
```

Comment:
The url is wrong.

I think we've given up on analytics for these docs and could just delete this.